### ISSUE TITLE
Decoding hyperparameter details

### ISSUE THREAD

**Issue by yuezih** [DAMO-NLP-SG/VCD#4]
Hi, great work and a really neat solution!

I'm encountering some problems when adopting VCD in LLaVA. When using the codebase you provided, I was unable to reproduce the results in the paper. I would like to ask you if you can provide more details to help me out :)

### Decoding hyperparameter
Here's the script I'm using:
```
python experiments/eval/object_hallucination_vqa_llava.py \
    --model-path llava-v1.5-7b \
    --question-file llava_pope_test.jsonl \
    --image-folder MSCOCO/images/val2014 \
    --answers-file answer-pope.jsonl \
    --conv-mode vicuna_v1 \
    --use_cd
```
With the script, the applied hyperparameters should be: 
`temperature=1, top_p=1, top_k=None`

However, by adding VCD, e.g., on MSCOCO popular, F1 value increases from 83.1 to 84.1, not as significant as the gains in the paper. Are my decoding hyperparameters not suitable?

### Sensitivity to temperature

BTW, I found that VCD seems to be sensitive to temperature, i.e. when I'm using a temperature T=0 (or 0.001), it reduces to regular decoding (with exactly the same results), or even leads to worse results than regular decoding. Do you have any idea about this?

---
Thank you very much!

**Comment by LengSicong** [comment]
Hi, thanks for your interest.
In our paper, all experiments are conducted and compared with "Direct Sampling." All improvements reported in our paper are based on direct sampling with/without VCD.
For temperature, may I know what is the experimental setting?

**Comment by yuezih** [comment]
Hi Sicong, I tried using a lower temperature in the POPE test because it seemed to result in better benchmark performance. However, when the temperature is very low, e.g., 0.001, VCD seems to behave exactly like the baseline, i.e., greedy search. I'm wondering if you have some insights about this phenomenon. Thanks a lot!

**Comment by LengSicong** [comment]
Hi, may you elaborate on the temperature more? Is it the temp used to normalize the output logits during sampling, or one of our hyper-parameters reported in the paper? Moreover, you can refer to our appendix for additional ablation studies on some sampling strategies other than direct sampling.

**Comment by yuezih** [comment]
It's the temperature used for normalizing logits. For example:
```
python experiments/eval/object_hallucination_vqa_llava.py \
    --model-path llava-v1.5-7b \
    --question-file llava_pope_test.jsonl \
    --image-folder MSCOCO/images/val2014 \
    --answers-file answer-pope.jsonl \
    --conv-mode vicuna_v1 \
    --temperature 0.001 \
    --use_cd
```
I'll read the appendix for more details. Thanks for your reply!

**Comment by LengSicong** [comment]
IMO, when the temperature is set to 0.001, the logits are all multiplied by 1000. If no other sampling strategy is applied, it is almost same as direct sampling.

**Comment by yuezih** [comment]
I see. Thanks!
